Added fave field to DB and scope to recipe model

rae-stanton opened the pull request:
This adds the ability to have a "favorite" recipe; default is false.

Also tested that boolean works in console.

Rspec tests that asserting true/false both work. 

Finally getting the hang of rspec hella yeah!

<img src="https://media2.giphy.com/media/jErnybNlfE1lm/giphy.gif"/>

rae-stanton committed:
Added fave field to DB and scope to recipe model

This adds the ability to have a "favorite" recipe; default is false.

Also tested that boolean works in console.

Rspec tests that asserting true/false both work.

spaulsandhu commented on app/models/recipe.rb:
```diff
@@ -11,5 +11,7 @@ class Recipe < ApplicationRecord
   validates :difficulty, presence: true
   validates :name, presence: true
+
+  scope :recipe_is_favorited, -> { where(favorite: true) }
```
For the name of this scope I don't think we need to call it `recipe_is_favorited` when we use the scope from the `User` model it will look like `user.recipes.recipe_is_favorited` which reads kind of strange. 

I think changing the name of the scope to something a little more simplified like `favorites` or `favorited` may read better ie. `user.recipes.favorites` or `user.recipes.favorited`.

spaulsandhu commented on spec/models/recipe_spec.rb:
```diff
@@ -2,7 +2,7 @@
 
 RSpec.describe Recipe, type: :model do
```
I'd love to see a spec that tests the scope too..... let me know if you want to chat about this.

spaulsandhu reviewed (commented):
Looks great! You blasted through this so fast. I left some ideas for renaming the scope and why, also, I'd like to challenge you to add a couple tests that assert the scope does what it says it does.

rae-stanton committed:
scope tests & scope naming (favorites)

Changed from `recipe_is_favorited` to `favorites` in recipe scope; added scope specific tests. (:

rae-stanton commented:
Added new rspec tests & changed name per your feedback. 
<img src="https://media3.giphy.com/media/VbnUQpnihPSIgIXuZv/giphy-downsized-medium.gif"/>

spaulsandhu reviewed (approved):
nice!
